spec/controllers/medications_controller_spec.rb

```diff
@@ -143,6 +143,9 @@ def post_create(medication_params)
         it 'has no validation errors' do
           post_create valid_medication_params
           expect(assigns(:medication).errors).to be_empty
+  
+          expect(assigns(:medication)[:refill]).to be_between("01-01-0001".to_date,
+            ((7 * assigns(:medication)[:total]) / (assigns(:medication)[:dosage] * assigns(:medication)[:weekly_dosage].count)).days.from_now)
```

Comment:
Is checking between 01-01-0001 and the number of days from when the days are calculated the best way of verifying this works? Why not set the created_at date to a value upon create and then check for the specific date in the test?

Reply:
![Screenshot from 2020-03-06 13-26-03](https://user-images.githubusercontent.com/32172764/76063939-b1ca6100-5fae-11ea-9338-598ff36dcc92.png)

Hi @julianguyen, I tried changing this part so that it checks whether the refill date is between date of creation of medication and the maximum possible date (as calculated). However this throws the above error. Could you please guide me further?

Reply:
Ah so this error is happening because the medication's refill is being set to `01/01/2020` in `spec/factories/factories.rb`. We use this factory to generate a medication in this test.

I tested this feature locally and it doesn't work because the front-end prevents us from submitting the form if the refill field is left blank.

Reply:
To fix this, we'll just need to set this field as non-required (just in the front-end not back-end). I can push a commit to this PR to update this.

Reply:
As for getting the test to pass, we'll need to call `add_refill` when we call `Medication.new` instead of doing it after the new medication entry is created. I'll also update your PR for you.